### PR TITLE
Fix minor typos for programer guide documents

### PR DESCRIPTION
**Pull request by imsheridan**
Described as PR title.

### CODE REVIEW

File: tensorflow/docs_src/programmers_guide/datasets.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ The `tf.data` API introduces two new abstractions to TensorFlow:
   element of a `Dataset` when executed, and typically acts as the interface
   between input pipeline code and your model. The simplest iterator is a
   "one-shot iterator", which is associated with a particular `Dataset` and
-  iterates through it once. For more sophisticated uses, the
+  iterates through it once. For more sophisticated users, the
```
Comment (mrry) [inline tensorflow/docs_src/programmers_guide/datasets.md]: This changes the sense of the sentence—please revert.

File: tensorflow/docs_src/programmers_guide/datasets.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ type of iterator, and the options are outlined below.
 
 ### Dataset structure
 
-A dataset comprises elements that each have the same structure. An element
+A dataset comprises elements that each has the same structure. An element
```
Comment (mrry) [inline tensorflow/docs_src/programmers_guide/datasets.md]: I prefer the previous version here—please revert.

**Review by mrry — CHANGES_REQUESTED** [review]
Please revert the changes in `datasets.md`.

File: tensorflow/docs_src/programmers_guide/low_level_intro.md [hunk 3]
```diff
@@ -323,7 +323,7 @@ The connection weights and biases are managed by the layer object.
 
 The following code creates a @{tf.layers.Dense$`Dense`} layer that takes a
 batch of input vectors, and produces a single output value for each. To apply a
-layer to an input, call the layer as if it were a function. For example:
+layer to an input, call the layer as if it was a function. For example:
```
Comment (frankchn) [inline tensorflow/docs_src/programmers_guide/low_level_intro.md]: Same thing with the past subjunctive, `were` is correct usage.

File: tensorflow/docs_src/programmers_guide/low_level_intro.md [hunk 4]
```diff
@@ -360,7 +360,7 @@ should be one of the last things added during graph construction.
 ### Executing Layers
 
 Now that the layer is initialized, we can evaluate the `linear_model`'s output
-tensor as we would any other tensor. For example, the following code:
+tensor as we could for any other tensor. For example, the following code:
```
Comment (frankchn) [inline tensorflow/docs_src/programmers_guide/low_level_intro.md]: `for` is not necessary, and either `would` or `could` is OK here.

**Review by frankchn — CHANGES_REQUESTED** [review]

File: tensorflow/docs_src/programmers_guide/low_level_intro.md [hunk 5]
```diff
@@ -115,7 +115,7 @@ operation that produces them followed by an output index, as in
 
 ### TensorBoard
 
-TensorFlow provides a utility called TensorBoard. One of TensorBoard's many
+TensorFlow provides an utility called TensorBoard. One of TensorBoard's many
```
Comment (frankchn) [inline tensorflow/docs_src/programmers_guide/low_level_intro.md]: `A` utility rather than `an`

**Review by frankchn — DISMISSED** [review]

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by imsheridan** [comment]
Is there any admin help review this patch?

**Review by mrry — APPROVED** [review]
